speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/Recorder.cpp

```diff
@@ -1,36 +1,40 @@
 #include "Recorder.h"
 #include <iomanip>
 #include <sstream>
-#include <iostream>
 #include <tgf.h>
 
 ///
 /// @brief Constructor of Recording, 
 /// creates a file with the current date and time,
 /// opens file in binary and appending mode
 /// initializes the previous input with 2's since user input is between -1 an 1.
-/// also creates a folder for the files
+/// also creates a folder for the files.
+///	@param p_dirName the name of the directory that needs to be placed in the test_data directory.
+///	@param p_fileName the name of the file placed in the directory,
+///	the current date will be added to this name
+///	@param p_paramAmount the amount of parameters that you want to save per line.
 /// @author Sjoerd Schilder
 ///
-Recorder::Recorder()
+Recorder::Recorder(const std::string& p_dirName, const std::string& p_fileName, const int p_paramAmount)
 {
 	// create directory if it doesn't exist
-	std::string dirName = "..\\test_data\\user_recordings";
-	if(!GfDirExists(dirName.c_str())) 
+	std::string dirPath = "..\\test_data\\" + p_dirName;
+	if(!GfDirExists(dirPath.c_str())) 
 	{
-		GfDirCreate(dirName.c_str());
+		GfDirCreate(dirPath.c_str());
 	}
 	
 	// create and open new recording file with current timestamp
 	std::time_t t = std::time(nullptr);
 	std::tm tm = *std::localtime(&t);
 	std::stringstream buffer;
 	buffer << std::put_time(&tm, "%Y%m%d-%H%M%S");
-	std::string fileName = dirName + "\\Record" + buffer.str() + ".txt";
-	m_recordingFile.open(fileName, std::ios::binary | std::ios::app);
+	std::string path = dirPath + "\\" + p_fileName + buffer.str() + ".txt";
```

Comment:
Not a huge problem but a suggestion, you could have a parameter that uses the [strftime](https://en.cppreference.com/w/c/chrono/strftime) format so the creator can decide on the file format.

Reply:
i don't think that isreally necesarry , maybe i something i could look at later